packages/auth/src/mfa/mfa_session.ts

```diff
@@ -33,17 +34,17 @@ export class MultiFactorSessionImpl implements MultiFactorSession {
   private constructor(
     readonly type: MultiFactorSessionType,
     readonly credential: string,
-    readonly auth?: AuthInternal
+    readonly user?: UserInternal
   ) {}
 
   static _fromIdtoken(
     idToken: string,
-    auth?: AuthInternal
+    user?: UserInternal
```

Comment:
I wonder if we can make "user" a required parameter here. The only place this is invoked without the "user" is the fromJSON method - https://github.com/firebase/firebase-js-sdk/blob/88f010a5b7c36c6ab1fef3df24a0210fc884b063/packages/auth/src/mfa/mfa_session.ts#L81

But I don't see this method being used anywhere. @sam-gc do you know why we have these methods?

Reply:
> I wonder if we can make "user" a required parameter here. The only place this is invoked without the "user" is the fromJSON method -
> 
> https://github.com/firebase/firebase-js-sdk/blob/88f010a5b7c36c6ab1fef3df24a0210fc884b063/packages/auth/src/mfa/mfa_session.ts#L81
> 
> But I don't see this method being used anywhere. Sam do you know why we have these methods?

toJSON can be invoked when an end user calls JSON.stringify(), but it's unclear where fromJSON is used.. since JSON.parse won't automatically call it and it is not in the expected reviver format - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/parse#using_the_reviver_parameter

But i think it is ok to leave it be, in case users are manually calling this function somewhere.